drivers/performance_measurement/app/models/performance_measurement/result_calculation.rb

```diff
@@ -202,115 +159,142 @@ def client_data(field, period, project_id: nil)
       @client_data.dig(key, project_id) || []
     end
 
-    def count_of_sheltered_homeless_clients(field, project: nil)
+    def count_of_sheltered_homeless_clients(detail, project: nil)
+      field = detail[:calculation_column]
       reporting_count = client_count(field, :reporting, project_id: project&.project_id)
       comparison_count = client_count(field, :comparison, project_id: project&.project_id)
 
+      goal_value = goal(detail[:goal_calculation])
+      progress = percent_changed(reporting_count, comparison_count)
```

Comment:
Maybe `passed?` should return the goal value and progress value as well so as to ensure that the reported value `secondary_value` always matches the one used in `passed?`
```passed, goal_value, progress = passed?(...```

Reply:
secondary value doesn't always match.  At the moment that's intentional, and why we need progress.  In the view, secondary value is (always/mostly?) %change year over year, but only some of the `passed` checks are across years.

Reply:
OK. It looked to me like `passed?` could always return the %change, or the second arg, if the third arg was `nil`.

Reply:
I'm going to make a `calculate_progress` method that accepts `goal_method`, `reporting_value`, and an optional `comparison_value` and returns `{ passed: boolean, goal: goal_value, progress: progress_value }`